src/python/pants/backend/python/macros/python_artifact.py

```diff
@@ -40,6 +70,38 @@ def binaries(self):
     def __str__(self) -> str:
         return self.name
 
+    @deprecated(
+        "2.6.0dev0",
+        """Use `python_distribution(entry_points={'console_scripts':{'<name>': '<entry
+point>'}})` instead of
+`python_distribution(provides=setup_py().with_binaries(...))`.
+
+The syntax for entry points must follow that of setuptools, and is specified as
+follows:
+
+    <name> = [<package>.[<subpackage>.]]<module>[:<object>.<object>]
+
+Example:
+
+    entry_points={
+      'console_scripts': {
+        'my_command': 'my.library.bin:main'
+      }
+    }
+
+The entry point must now be provided explicitly and are not derived from a
+`pex_binary` target.
```

Comment:
If you think that we should support pex_binary addresses via John's unambiguous syntax, it would be great to include in this PR so that this deprecation can be fixed.

Reply:
dropped deprecation for later.